ZEngine/src/VulkanInstance.cpp

```diff
@@ -54,7 +54,8 @@ namespace ZEngine::Hardwares
         // std::copy(std::begin(validation_layer_name_collection), std::end(validation_layer_name_collection), std::back_inserter(instance_layer_name_collection));
 #endif
 
-        std::unordered_set<std::string> extension_layer_name_collection = {VK_KHR_SURFACE_EXTENSION_NAME, VK_KHR_WIN32_SURFACE_EXTENSION_NAME, VK_EXT_DEBUG_UTILS_EXTENSION_NAME};
+        std::unordered_set<std::string> extension_layer_name_collection = {
+            VK_KHR_SURFACE_EXTENSION_NAME, VK_KHR_WIN32_SURFACE_EXTENSION_NAME, VK_EXT_DEBUG_UTILS_EXTENSION_NAME};
 
         auto layer_properties = m_layer.GetInstanceLayerProperties();
         for (const LayerProperty& layer : layer_properties)
```

Comment:
This is not how you should choose layers to enable. Depending on the machine, enabling all available layers and associated extensions might not even work since they may not be compatible. As a practical example, with this code the instance creation fails due to an unsupported layer configuration.

What you rather should do is decided beforehand which layers you need for your application, such as VK_LAYER_LUNARG_api_dump that you specify above, verify if the instance actually supports the layer using GetInstanceLayerProperties, and finally create the instance with only the layers you need and are supported.

Reply:
Thanks !!
I updated the implementation,
lemme know if it fixes the crash on your side

Reply:
Still fails instance creation. Did you push your changes? I pulled from vulkan-pipeline but to me it looks like the same code is still in there, meaning you still enable all instance layers that get returned by GetInstanceLayerProperties.

Reply:
Revisited the impl., by selecting the required layers, hope that fixes the crash from your side 🤞

Reply:
Doh !🤦my bad.. pushed now ... thanks !!!